openfecli/commands/gather_abfe.py

```diff
@@ -30,17 +31,15 @@ def _get_name(result: dict) -> str:
     str
         Ligand name corresponding to the results.
     """
-    try:
-        nm = list(result["unit_results"].values())[0]["name"]
 
-    except KeyError:
-        raise ValueError("Failed to guess name")
+    # TODO: do we always want to treat "solvent" as the source of truth? should we verify that "complex" has the same name?
+    # TODO: is it faster to only load the ligands, or is loading the entire pur cheap?
+    solvent_data = list(result["protocol_result"]["data"]["solvent"].values())[0][0]
+
+    pur_solvent = gufe.ProtocolUnitResult.from_dict(solvent_data)
+
+    name = pur_solvent.inputs["stateA"].components["ligand"].name
```

Comment:
I _think_ this is a bad footgun! I don't think the "ligand" key is guaranteed.

Reply:
ah, makes sense! A couple other ways we could do this:

1. `name = pur_solvent.inputs['alchemical_components']['stateA'][0].name`: assumes we want the name of the first entry of `stateA`
3. `name = pur_solvent.inputs['stateA'].get_components_of_type(gufe.SmallMoleculeComponent)[0].name`: explicitly pulls `SmallMoleculeComponent`: I see this is a restriction of ABFE's currently, but I'd prefer to stay more flexible.

Also, we can anticipate multiple entries by joining them - @IAlibay is this something that should be valid?
` name  = "_".join([i.name for i in pur_solvent.inputs['alchemical_components']['stateA']])`

Reply:
Same thing for SepTop, you can just get name A and name B by looking at the entry for each alchemical component.

Reply:
done! I assume I should use `alchemical_components` for septop as well?

Reply:
I believe so, yes. It has the same kind of validation checks if I remember correctly.